extensions/vscode/src/views/deployProgress.ts

```diff
@@ -9,32 +9,54 @@ export function deployProject(localID: string, stream: EventStream) {
     {
       location: ProgressLocation.Notification,
       title: `Deploying your project`,
-      cancellable: false,
+      cancellable: true,
     },
-    (progress) => {
+    (progress, token) => {
       let resolveCB: (reason?: any) => void;
       let rejectCB: (reason?: any) => void;
       const registrations: UnregisterCallback[] = [];
+      const promise = new Promise<void>((resolve, reject) => {
+        resolveCB = resolve;
+        rejectCB = reject;
+      });
+      let streamID = localID;
 
-      const unregiserAll = () => {
+      const unregisterAll = () => {
         registrations.forEach((cb) => cb.unregister());
       };
 
-      const promise = new Promise<void>((resolve, reject) => {
-        resolveCB = resolve;
-        rejectCB = reject;
+      token.onCancellationRequested(() => {
+        streamID = "NEVER_A_VALID_STREAM";
```

Comment:
I'm a bit confused why this setting of `streamID` is here.

Reply:
Just to make sure it NEVER matches when compared against incoming streamIDs, since there are still events flowing through the pipes.